File: mergeTwoSortedArrays.js
// Description
// Merge two given sorted integer array A and B into a new sorted integer array.

// Example
// A = [1, 2, 3, 4]
// B = [2, 4, 5, 6]
// return [1, 2, 2, 3, 4, 4, 5, 6]

/**
 * @param A: sorted integer array A
 * @param B: sorted integer array B
 * @return: A new sorted integer array
 */
const mergeSortedArray = function (A, B) {
  let left = 0
  let right = 0
  let C = []
  while (left < A.length && right < B.length) {
    if (A[left] < B[right]) {
      C.push(A[left])
      left++
    } else {
      C.push(B[right])
      right++
    }
  }
  if (left < A.length) {
    let t = A.slice(left)
    C = C.concat(t)
  }
  if (right < B.length) {
    let t = B.slice(right)
    C = C.concat(t)
  }
  return C
}

File: mergeTwoSortedArrays.py
class Solution:
    """
    @param A: sorted integer array A
    @param B: sorted integer array B
    @return: A new sorted integer array
    """
    def mergeSortedArray(self, A, B):
        # write your code here
        C = []
        left, right = 0, 0
        while left < len(A) and right < len(B):
            if A[left] < B[right]:
                C.append(A[left])
                left += 1
            else:
                C.append(B[right])
                right += 1
        if left < len(A):
            for _ in xrange(left, len(A)):
                C.append(A[_])
        if right < len(B):
            for _ in xrange(right, len(B)):
                C.append(B[_])
        return C